iphone/Classes/TiUIWebView.m

```diff
@@ -1010,20 +1021,37 @@ - (void)webView:(WKWebView *)webView decidePolicyForNavigationAction:(nonnull WK
   }
   NSArray<NSString *> *allowedURLSchemes = [[self proxy] valueForKey:@"allowedURLSchemes"];
 
-  // Handle blacklisted URL's
+  // Do not load the URL if it's black-listed.
+  // DEPRECATED: Should use "blockedURLs" property with a "blockedurl" event listener instead.
+  NSString *urlCandidate = navigationAction.request.URL.absoluteString;
   if (_blacklistedURLs != nil && _blacklistedURLs.count > 0) {
-    NSString *urlCandidate = navigationAction.request.URL.absoluteString;
-
-    for (NSString *blackListedURL in _blacklistedURLs) {
-      if ([urlCandidate rangeOfString:blackListedURL options:NSCaseInsensitiveSearch].location != NSNotFound) {
+    for (NSString *blockedURL in _blacklistedURLs) {
+      if ([urlCandidate rangeOfString:blockedURL options:NSCaseInsensitiveSearch].location != NSNotFound) {
         if ([[self proxy] _hasListeners:@"blacklisturl"]) {
           [[self proxy] fireEvent:@"blacklisturl"
                        withObject:@{
                          @"url" : urlCandidate,
                          @"message" : @"Webview did not load blacklisted url."
                        }];
         }
+        decisionHandler(WKNavigationActionPolicyCancel);
+        [self _cleanupLoadingIndicator];
+        return;
+      }
+    }
+  }
 
+  // Do not load the URL if it's on the block-list.
+  if (_blockedURLs != nil && _blockedURLs.count > 0) {
+    for (NSString *blockedURL in _blockedURLs) {
+      if ([urlCandidate rangeOfString:blockedURL options:NSCaseInsensitiveSearch].location != NSNotFound) {
+        if ([[self proxy] _hasListeners:@"blockedurl"]) {
+          [[self proxy] fireEvent:@"blockedurl"
+                       withObject:@{
+                         @"url" : urlCandidate,
+                         @"message" : @"Webview did not load blocked url."
+                       }];
+        }
         decisionHandler(WKNavigationActionPolicyCancel);
         [self _cleanupLoadingIndicator];
         return;
```

Comment:
I would be better to merge 'backlisteurl' event related code snippet inside 'blockedurl' . So above code snippet can be replaced by following-
`  // Do not load the URL if it's on the block-list.
  if (_blockedURLs != nil && _blockedURLs.count > 0) {
    NSString *urlCandidate = navigationAction.request.URL.absoluteString;
    for (NSString *blockedURL in _blockedURLs) {
      if ([urlCandidate rangeOfString:blockedURL options:NSCaseInsensitiveSearch].location != NSNotFound) {
        if ([[self proxy] _hasListeners:@"blacklisturl"]) {
          DEPRECATED_REPLACED(@"UI.WebView.blacklisturl", @"9.2.0", @"UI.WebView.blockedurl");

          [[self proxy] fireEvent:@"blacklisturl"
                       withObject:@{
                         @"url" : urlCandidate,
                         @"message" : @"Webview did not load blacklisted url."
                       }];
        }
        if ([[self proxy] _hasListeners:@"blockedurl"]) {
          [[self proxy] fireEvent:@"blockedurl"
                       withObject:@{
                         @"url" : urlCandidate,
                         @"message" : @"Webview did not load blocked url."
                       }];
        }
        decisionHandler(WKNavigationActionPolicyCancel);
        [self _cleanupLoadingIndicator];
        return;`

Reply:
I separated the 2 on purpose so that we can easily delete the "blacklisted" code in the future.
(Otherwise I wouldn't normally duplicate code like this.)

So, the way it works now is a "blacklisturl" event will only be fired for matches in the `blacklistedURLs` property... and a "blockedurl" event will only be fired for matches in the `blockedURLs` property. I purposely treat them separately on both Android and iOS.

Reply:
Thanks for explaining. Approved!